From 3804648d98b409c60bedd9a53ab4e370e1005ec1 Mon Sep 17 00:00:00 2001
From: Larry Finger <Larry.Finger@lwfinger.net>
Date: Sat, 22 Jun 2013 22:03:24 -0500
Subject: [PATCH 166/210] rtl8192du: Fix problem with disabling
 CONFIG_CONCURRENT_MAC

In the original form, an oops resulted whenever the configuration parameter
of the title was disabled. The downside was that 4 wlanX units were created
rather than one 2.4 and one 5 GHz unit.

This commit also removes that configuration parameter.

Signed-off-by: Larry Finger <Larry.Finger@lwfinger.net>
---
 hal/hal_intf.c     | 29 +++++------------------------
 include/autoconf.h |  2 +-
 2 files changed, 6 insertions(+), 25 deletions(-)

--- a/drivers/staging/rtl8192du/hal/hal_intf.c
+++ b/drivers/staging/rtl8192du/hal/hal_intf.c
@@ -84,17 +84,16 @@ uint	 rtw_hal_init(struct rtw_adapter *p
 {
 	uint	status = _SUCCESS;
 
-	if (padapter->hw_init_completed == true)
-	{
+	if (padapter->hw_init_completed == true) {
 		DBG_8192D("rtw_hal_init: hw_init_completed == true\n");
 		return status;
 	}
 
 #ifdef CONFIG_DUALMAC_CONCURRENT
 	/*  before init mac0, driver must init mac1 first to avoid usb rx error. */
-	if ((padapter->pbuddy_adapter != NULL) && (padapter->DualMacConcurrent == true)
-		&& (padapter->adapter_type == PRIMARY_ADAPTER))
-	{
+	if ((padapter->pbuddy_adapter != NULL) &&
+	    (padapter->DualMacConcurrent == true) &&
+	    (padapter->adapter_type == PRIMARY_ADAPTER)) {
 		if (padapter->pbuddy_adapter->hw_init_completed == true) {
 			DBG_8192D("rtw_hal_init: pbuddy_adapter hw_init_completed == true\n");
 		} else {
@@ -108,20 +107,6 @@ uint	 rtw_hal_init(struct rtw_adapter *p
 			}
 		}
 	}
-#else
-	if (adapter_to_dvobj(padapter)->NumInterfaces == 2 && padapter->registrypriv.mac_phy_mode != 1) {
-		if (WARN_ON_ONCE(!padapter->pbuddy_adapter))
-			return _FAIL;
-		if (padapter->pbuddy_adapter->hw_init_completed == false) {
-			status = padapter->HalFunc.hal_init(padapter->pbuddy_adapter);
-			if (status == _SUCCESS) {
-				padapter->pbuddy_adapter->hw_init_completed = true;
-			} else {
-				padapter->pbuddy_adapter->hw_init_completed = false;
-				RT_TRACE(_module_hal_init_c_,_drv_err_,("rtw_hal_init: hal__init fail for another interface\n"));
-			}
-		}
-	}
 #endif
 
 	padapter->hw_init_completed=false;
@@ -153,14 +138,10 @@ _func_enter_;
 
 	status = padapter->HalFunc.hal_deinit(padapter);
 
-	if (status == _SUCCESS) {
+	if (status == _SUCCESS)
 		padapter->hw_init_completed = false;
-	}
 	else
-	{
 		RT_TRACE(_module_hal_init_c_,_drv_err_,("\n rtw_hal_deinit: hal_init fail\n"));
-	}
-
 _func_exit_;
 
 	return status;
--- a/drivers/staging/rtl8192du/include/autoconf.h
+++ b/drivers/staging/rtl8192du/include/autoconf.h
@@ -104,7 +104,7 @@
 #endif	/*  CONFIG_BR_EXT */
 
 
-#define CONFIG_CONCURRENT_MODE 1
+//#define CONFIG_CONCURRENT_MODE 1
 #ifdef CONFIG_CONCURRENT_MODE
 	#define CONFIG_TSF_RESET_OFFLOAD 1			/*  For 2 PORT TSF SYNC. */
 #endif	/*  CONFIG_CONCURRENT_MODE */
